api/external_reviews_api.py

```diff
@@ -0,0 +1,246 @@
+  STAGE_BLINK_SHIPPING: StageType.SHIPPING,
+  STAGE_FAST_PROTOTYPE: StageType.PROTOTYPING,
+  STAGE_FAST_DEV_TRIAL: StageType.DEV_TRIAL,
+  STAGE_FAST_ORIGIN_TRIAL: StageType.ORIGIN_TRIAL,
+  STAGE_FAST_EXTEND_ORIGIN_TRIAL: StageType.ORIGIN_TRIAL,
+  STAGE_FAST_SHIPPING: StageType.SHIPPING,
+  STAGE_PSA_IMPLEMENT: StageType.WIDE_REVIEW,
+  STAGE_PSA_DEV_TRIAL: StageType.DEV_TRIAL,
+  STAGE_PSA_SHIPPING: StageType.SHIPPING,
+  STAGE_DEP_PLAN: StageType.INCUBATING,
+  STAGE_DEP_DEV_TRIAL: StageType.DEV_TRIAL,
+  STAGE_DEP_SHIPPING: StageType.SHIPPING,
+  STAGE_DEP_DEPRECATION_TRIAL: StageType.SHIPPED,
+  STAGE_DEP_EXTEND_DEPRECATION_TRIAL: StageType.SHIPPED,
+  STAGE_DEP_REMOVE_CODE: StageType.SHIPPED,
+  STAGE_ENT_ROLLOUT: StageType.SHIPPED,
+  STAGE_ENT_SHIPPED: StageType.SHIPPED,
+}
+
+
+def stage_type(feature: FeatureEntry, stage: Stage | None) -> StageType:
+  if stage is None:
+    return StageType.INCUBATING
+
+  tentative_type = STAGE_TYPES[stage.stage_type]
+  if (
+    tentative_type == StageType.SHIPPING
+    and feature.impl_status_chrome == ENABLED_BY_DEFAULT
+  ):
+    return StageType.SHIPPED
+  return tentative_type
+
+
+def min_of_present(*args):
+  present_args = [arg for arg in args if arg is not None]
+  if len(present_args) == 0:
+    return None
+  return min(present_args)
+
+
+def max_of_present(*args):
+  present_args = [arg for arg in args if arg is not None]
+  if len(present_args) == 0:
+    return None
+  return max(present_args)
+
+
+class ExternalReviewsAPI(basehandlers.APIHandler):
+  """Implements the OpenAPI /external_reviews path."""
+
+  def do_get(self, **kwargs):
+    """Get a list of features with outstanding external reviews from a particular review body.
+    """
+    review_group: str | None = kwargs.get('review_group', None)
+    if review_group not in ['tag', 'gecko', 'webkit']:
+      self.abort(404, f'invalid review group {review_group}')
+
+    # The name of the field in FeatureEntry that holds the review link for review_group.
+    review_link: str
+    # Matches URLs in the reviewer's review repository.
+    review_pattern: re.Pattern
+
+    # Get all the features that might have an unfinished review for the requested group.
+    if review_group == 'tag':
+      unreviewed_features = FeatureEntry.query(
+        FeatureEntry.has_open_tag_review == True  # noqa: E712
+      ).fetch()
+      review_link = 'tag_review'
+      review_pattern = TAG_REVIEW_URL_PATTERN
+    elif review_group == 'gecko':
+      unreviewed_features = FeatureEntry.query(
+        FeatureEntry.has_open_ff_review == True  # noqa: E712
+      ).fetch()
+      review_link = 'ff_views_link'
+      review_pattern = GECKO_REVIEW_URL_PATTERN
+    elif review_group == 'webkit':
+      unreviewed_features = FeatureEntry.query(
+        FeatureEntry.has_open_safari_review == True  # noqa: E712
+      ).fetch()
+      review_link = 'safari_views_link'
+      review_pattern = WEBKIT_REVIEW_URL_PATTERN
```

Comment:
If we anticipate adding more review groups in the future, consider moving this section into a separate helper method to help prevent this from growing too long.